gadopt/approximations.py

```diff
@@ -335,6 +347,11 @@ def __init__(self,
         self.Tbar = Tbar
         self.cp = cp
 
+    def stress(self, u):
+        stress = super().stress(u)
+        dim = len(u)  # Geometric dimension, i.e. 2D or 3D
```

Comment:
Is this always the same as `u.function_space().mesh().geometric_dimension()` (which the previous definition used)? Other than that, it's curious that there ends up being a big difference in the TALA DG benchmark...